sdk/python/feast/feature_store.py

```diff
@@ -378,6 +378,7 @@ def apply(
                 ]
             ],
         ],
+        update_infra: bool = True,
```

Comment:
Do we need to add a new flag here? Can't we just ensure that it's `FeatureStore` that's calling `update_infra` and it's not called by `repo_operations` instead? I'd really prefer to not introduce branching behaviour here unless absolutely necessary.